frontend/src/store/features/staking/stakeSlice.ts

```diff
@@ -353,6 +357,83 @@ export const txUnDelegate = createAsyncThunk(
   }
 );
 
+export const txCancelUnbonding = createAsyncThunk(
+  'staking/cancel-unbonding',
+  async (
+    data: TxCancelUnbondingInputs,
+    { rejectWithValue, fulfillWithValue, dispatch }
+  ) => {
+    try {
+      const msg = Unbonding(
+        data.delegator,
+        data.validator,
+        data.amount,
+        data.denom,
+        data.creationHeight
+      );
+      msg.value.creationHeight = msg.value.creationHeight.toString();
+      const result = await signAndBroadcast(
+        data.basicChainInfo.chainID,
+        data.basicChainInfo.aminoConfig,
+        data.basicChainInfo.prefix,
+        [msg],
+        GAS_FEE,
+        '',
+        `${data.feeAmount}${data.denom}`,
+        data.basicChainInfo.rest,
+        data.feegranter?.length > 0 ? data.feegranter : undefined
+      );
+      const tx = NewTransaction(
+        result,
+        [msg],
+        data.basicChainInfo.chainID,
+        data.basicChainInfo.address
+      );
+
+      dispatch(
+        addTransactions({
+          chainID: data.basicChainInfo.chainID,
+          address: data.basicChainInfo.cosmosAddress,
+          transactions: [tx],
+        })
+      );
+      dispatch(setTxAndHash({ tx, hash: tx.transactionHash }));
+
+      if (result?.code === 0) {
+        const inputData = {
+          baseURL: data.basicChainInfo.baseURL,
+          address: data.delegator,
+          chainID: data.basicChainInfo.chainID,
+        };
+        dispatch(resetDelegations({ chainID: data.basicChainInfo.chainID }));
+        dispatch(getDelegations(inputData));
+        dispatch(getUnbonding(inputData));
+        return fulfillWithValue({ txHash: result?.transactionHash });
+      } else {
+        return rejectWithValue(result?.rawLog);
+      }
+    } catch (error) {
+      console.log('err in catch ', error);
```

Comment:
It's important to have proper console messages so that we can easily track errors  "Error while cancel unbonding the transaction" in the catch block.